orchestrator/careplancontributor/integration_test.go

```diff
@@ -0,0 +1,213 @@
+		var fetchedTask fhir.Task
+		err := cpcDataRequester.Read("Task/"+*task.Id, &fetchedTask)
+		require.Error(t, err)
+	}
+	t.Log("Reading task after accepted - invalid principal - Fails")
+	{
+		cpcDataRequester := fhirclient.New(cpcURL, &http.Client{Transport: invalidTestPrincipalTransport}, nil)
+		var fetchedTask fhir.Task
+		err := cpcDataRequester.Read("Task/"+*task.Id, &fetchedTask)
+		require.Error(t, err)
+	}
+}
+
+func setupIntegrationTest(t *testing.T, notificationEndpoint *url.URL) (*url.URL, *httptest.Server, *url.URL) {
+	fhirBaseURL := setupHAPI(t)
+	config := careplanservice.DefaultConfig()
+	config.Enabled = true
+	config.FHIR.BaseURL = fhirBaseURL.String()
+	service, err := careplanservice.New(config, profile.TestProfile{
+		TestCsdDirectory: profile.TestCsdDirectory{Endpoint: notificationEndpoint.String()},
+	}, orcaPublicURL)
+	require.NoError(t, err)
+
+	serverMux := http.NewServeMux()
+	httpService := httptest.NewServer(serverMux)
+	service.RegisterHandlers(serverMux)
+
+	carePlanServiceURL, _ := url.Parse(httpService.URL + "/cps")
+
+	cpcConfig := DefaultConfig()
+	cpcConfig.Enabled = true
+	cpcConfig.FHIR.BaseURL = fhirBaseURL.String()
+	cpcConfig.CarePlanService.URL = carePlanServiceURL.String()
+	sessionManager, _ := createTestSession()
+	cpc, err := New(cpcConfig, profile.TestProfile{}, orcaPublicURL, sessionManager)
+	require.NoError(t, err)
+
+	cpcServerMux := http.NewServeMux()
+	cpcHttpService := httptest.NewServer(cpcServerMux)
+	cpc.RegisterHandlers(cpcServerMux)
+	cpcURL, _ := url.Parse(cpcHttpService.URL + "/contrib/fhir")
+
+	// Return the URLs and httpService of the cpsDataRequester, cpcDataRequester, cpsDataHolder
+	// these will be used to construct clients with both the correct and incorrect auth for positive and negative testing
+	return carePlanServiceURL, httpService, cpcURL
+}
+
+func setupNotificationEndpoint(t *testing.T) *url.URL {
+	notificationEndpoint := httptest.NewServer(http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+		notificationCounter.Add(1)
+		writer.WriteHeader(http.StatusOK)
+	}))
+	t.Cleanup(func() {
+		notificationEndpoint.Close()
+	})
+	u, _ := url.Parse(notificationEndpoint.URL)
+	return u
+}
+
+func setupHAPI(t *testing.T) *url.URL {
+	ctx := context.Background()
+	req := testcontainers.ContainerRequest{
+		Image:        "hapiproject/hapi:v7.2.0",
+		ExposedPorts: []string{"8080/tcp"},
+		Env: map[string]string{
+			"hapi.fhir.fhir_version": "R4",
+		},
+		WaitingFor: wait.ForHTTP("/fhir/Task"),
+	}
+	container, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	})
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		if err := container.Terminate(ctx); err != nil {
+			panic(err)
+		}
+	})
+	endpoint, err := container.Endpoint(ctx, "http")
+	require.NoError(t, err)
+	u, err := url.Parse(endpoint)
+	require.NoError(t, err)
+	return u.JoinPath("fhir")
+}
```

Comment:
I think this is duplicate now, maybe we move it to somewhere, where both integration tests can use it?

Reply:
Sounds good, will do